shell/platform/windows/platform_handler.cc

```diff
@@ -354,48 +366,65 @@ void PlatformHandler::SystemSoundPlay(
 }
 
 void PlatformHandler::SystemExitApplication(
-    const std::string& exit_type,
-    int64_t exit_code,
+    AppExitType exit_type,
+    UINT exit_code,
     std::unique_ptr<MethodResult<rapidjson::Document>> result) {
   rapidjson::Document result_doc;
   result_doc.SetObject();
-  if (exit_type.compare(kExitTypeRequired) == 0) {
-    QuitApplication(exit_code);
+  if (exit_type == AppExitType::required) {
+    QuitApplication(exit_code, nullptr);
```

Comment:
Oh wow, I didn't realize `HWND` was optional until very late in the review. This is not at all obvious and will result in crashes if folks miss this. Please make all these `HWND`s `std::optional<HWND>`s